### PR TITLE
fix(clerk-js): Fix optional field bug in user profile

### PR DESCRIPTION
**Pull request by raptisj**
## Type of change

- [x] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend-core`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/edge`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

Fix optional field bug in user profile

### CODE REVIEW

File: packages/clerk-js/src/ui/components/UserProfile/ProfilePage.tsx [hunk 1]
```diff
@@ -17,6 +17,8 @@ export const ProfilePage = withCardStateProvider(() => {
   const { first_name, last_name } = useEnvironment().userSettings.attributes;
   const showFirstName = first_name.enabled;
   const showLastName = last_name.enabled;
+  const userFirstName = user.firstName || '';
+  const userLastName = user.lastName || '';
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/components/UserProfile/ProfilePage.tsx:21]: Lets inline all these variables as they don't really benefit as much, eg 
```suggestion
  const { first_name: firstName, last_name: lastName } = useEnvironment().userSettings.attributes;
  
  ...
  // use firstName.enable, firstName.required and user.firstName directly
```

File: packages/clerk-js/src/ui/components/UserProfile/ProfilePage.tsx [hunk 2]
```diff
@@ -32,9 +34,12 @@ export const ProfilePage = withCardStateProvider(() => {
   });
 
   const userInfoChanged =
-    (showFirstName && firstNameField.value && firstNameField.value !== user.firstName) ||
-    (showLastName && lastNameField.value && lastNameField.value !== user.lastName);
-  const canSubmit = userInfoChanged || avatarChanged;
+    (showFirstName && firstNameField.value !== userFirstName) || (showLastName && lastNameField.value !== userLastName);
+  const canSubmitOptionalFields = userInfoChanged || avatarChanged;
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/components/UserProfile/ProfilePage.tsx:38]:
```suggestion
  const fieldsChanged = userInfoChanged || avatarChanged;
```

File: packages/clerk-js/src/ui/components/UserProfile/ProfilePage.tsx [hunk 3]
```diff
@@ -32,9 +34,12 @@ export const ProfilePage = withCardStateProvider(() => {
...
+
+  const hasRequiredFields = (showFirstName && first_name.required) || (showLastName && last_name.required);
+  const canSubmitRequiredFields =
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/components/UserProfile/ProfilePage.tsx:41]:
```suggestion
  const requiredFieldsFilled =
```

File: packages/clerk-js/src/ui/components/UserProfile/ProfilePage.tsx [hunk 4]
```diff
@@ -61,19 +66,19 @@ export const ProfilePage = withCardStateProvider(() => {
...
             <Form.ControlRow>
               <Form.Control
                 {...lastNameField.props}
-                required
+                required={last_name.required}
               />
             </Form.ControlRow>
           )}
-          <FormButtons isDisabled={!canSubmit} />
+          <FormButtons isDisabled={hasRequiredFields ? !canSubmitRequiredFields : !canSubmitOptionalFields} />
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/components/UserProfile/ProfilePage.tsx:81]:
```suggestion
          <FormButtons isDisabled={fieldsChanged && requiredFieldsFIlled} />
```

**Review by nikosdouvlis — COMMENTED** [review]
Some minor changes, otherwise looking good!

File: packages/clerk-js/src/ui/components/UserProfile/ProfilePage.tsx (hunk 1 above)
Comment (raptisj) [reply packages/clerk-js/src/ui/components/UserProfile/ProfilePage.tsx:21]: @nikosdouvlis we need to have empty strings as default values(e.x. `  const userLastName = user.lastName || '';`) in order to make proper check later on(e.x. `lastNameField.value !== userLastName`).

**Review by SokratisVidros — APPROVED** [review]

**Pull request merged**